sycl/doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc

```diff
@@ -0,0 +1,304 @@
+
+To be aligned with the SYCL 2020 group algorithms, an additional group argument is added to the matrix operations to designate that these functions are collective operations. The {dpcpp} syntax is the following: 
+
+IMPORTANT: In the current implementation, only the subgroup scope is supported
+
+#### Load 
+```c++
+namespace sycl::ext::intel::experimental::matrix {
+  template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix_layout Layout,
+          access::address_space Space>
+  void joint_matrix_load(Group sg, joint_matrix<Group, T, NumRows, NumCols> &res,
+		    multi_ptr<T, Space> src, size_t stride, matrix_layout layout = matrix_layout::row_major);
+}
+```
+This function loads data from memory to the 2d tiles of AMX that is a 2d storage.
+
+
+#### Store 
+```c++
+namespace sycl::ext::intel::experimental::matrix {
+  template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix_layout Layout,
+          access::address_space Space>	  
+  void joint_matrix_store(Group sg, matrix<Group, T, NumRows, NumCols> &res,
+		     multi_ptr<T, Space> src, size_t stride, matrix_layout layout = matrix_layout::row_major);
+}
+```
+This function stores the data from the 2d tiles back to memory.
+
+#### Multiply and Add
+
+```c++
+namespace sycl::ext::intel::experimental::matrix {
+  template <typename Group, typename T1, typename T2, std::size_t M,
+          std::size_t K, std::size_t N,
+	  matrix_layout LayoutA, matrix_layout LayoutB,
+          matrix_layout LayoutC>
+  joint_matrix<Group, T2, M, N> joint_matrix_mad(Group sg, joint_matrix<Group, T1, M, K> A,
+               joint_matrix<Group, T1, K, N> B, joint_matrix<Group, T2, M, N> C);
+}
+```
+The matrix multiply and add function performs the multiply operation on the matrices `A` and `B`, accumulate the result with `C` and return the result.
+
+
+## VNNI/Packed Layout
+AMX compute assumes register for B tile (src1) to be in VNNI format as they need 32bit of K-data in A and B to be contiguous in memory.
+The VNNI blocking factor is 2 in the case of 16bits, 4 in the case of 8 bits elements. While the current implementation assumes that the matrix has been already packed by the user for performance reasons, the layout information is needed to inform the implementation about this transform.
+
+
+      // Example of bf16 data type: For the 4 rows x 4 columns matrix below
+      // ---------------------------------
+      // a1, b1, c1, d1
+      // a2, b2, c2, d2
+      // a3, b3, c3, d3
+      // a4, b4, c4, d4
+      // ---------------------------------
+      // The reformating to VNNI layout yield the below 2 rows x 8 columns matrix
+      // ---------------------------------
+      // a1, a2, b1, b2, c1, c2, d1, d2
+      // a3, a4, b3, b4, c3, c4, d3, d4
+
```

Comment:
It might be useful to illustrate the same thing for an 8-bit type.

Reply:
Yes, I will add an example for that